src/app/shared/components/autocomplete/autocomplete.component.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  Component,
+  Input,
+  OnInit,
+  forwardRef,
+  OnDestroy,
+  HostBinding,
+  Injector,
+  Type
+} from '@angular/core';
+import {
+  ControlValueAccessor,
+  FormControl,
+  FormGroup,
+  NG_VALUE_ACCESSOR,
+  NgControl
+} from '@angular/forms';
+import { FormFieldControl } from '@app/shared/components/control';
+import { Subject } from 'rxjs';
+import { takeUntil, tap } from 'rxjs/operators';
+
+@Component({
+  selector: 'vtm-autocomplete',
+  templateUrl: './autocomplete.component.html',
+  styleUrls: ['./autocomplete.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => AutocompleteComponent),
+      multi: true
+    },
+    {
+      provide: FormFieldControl,
+      useExisting: AutocompleteComponent
+    }
+  ]
+})
+export class AutocompleteComponent
+  implements ControlValueAccessor, FormFieldControl, OnInit, OnDestroy {
+  @Input() autocompleteData;
+  @Input('aria-describedby') ariaDescribedBy: string | null;
+  autocompleteInput: FormControl;
+  errorMessages?: { rule: string; message: string }[];
+  controlType = 'text';
+  multi = true;
+
+  @Input('value') _value = '';
```

Comment:
Can we please group the @input together into same logical groups?
It will make readability and understanding of the component easier, we could have @input, @output, 'private vars', angular lifecycles, etc..